rockcraft/plugins/python_plugin.py

```diff
@@ -85,20 +85,35 @@ def _get_system_python_interpreter(self) -> Optional[str]:
 
     @override
     def _get_script_interpreter(self) -> str:
-        """Overridden because Python is always available in /bin/python3."""
-        return "#!/bin/python3"
+        """Overridden because Python is always available in /bin."""
+        return "#!/bin/${PARTS_PYTHON_INTERPRETER}"
 
     @override
     def get_build_commands(self) -> List[str]:
         """Overridden to add a sitecustomize.py ."""
-        original = super().get_build_commands()
+        commands = []
+
+        # Detect whether PARTS_PYTHON_INTERPRETER is a full path (not supported)
+        commands.append(
+            dedent(
+                """
+                # Detect whether PARTS_PYTHON_INTERPRETER is a full path
+                if [[ "${PARTS_PYTHON_INTERPRETER}" = /* ]]; then
+                    echo "Full paths in \"PARTS_PYTHON_INTERPRETER\" are not allowed: ${PARTS_PYTHON_INTERPRETER}"
```

Comment:
Maybe absolute paths instead of full paths?

Reply:
argh you are so right. I'll fix